rootfs-builder/rootfs.sh

```diff
@@ -455,10 +455,16 @@ prepare_overlay()
 {
 	pushd "${ROOTFS_DIR}" > /dev/null
 	mkdir -p ./etc ./lib/systemd ./sbin ./var
-	ln -sf  ./usr/lib/systemd/systemd ./init
-	ln -sf  ../../init ./lib/systemd/systemd
-	ln -sf  ../init ./sbin/init
-	# Kata sytemd unit file
+	if [ ! -e ./sbin/init ]; then
+		# This symlink hacking is mostly to make later rootfs
+		# validation work correctly for the dracut case.
+		# We skip this if /init exists in the rootfs, meaning
```

Comment:
*if /init exist* ?  or if *sbin/init exists*, sorry I don't understand this

Reply:
Updated the comment to mention /sbin/init and moved it to hopefully improve the context. If you meant you don't understand the patch, please confirm and I will provide more info (the usage of this function has some subtle semantic differences depending on whether and how dracut is used with it)